src/producer/updater/MySegments.js

```diff
@@ -32,24 +32,34 @@ function MySegmentsUpdaterFactory(context) {
   let readyOnAlreadyExistentState = true;
   let startingUp = true;
 
-  return function MySegmentsUpdater(retry = 0) {
+  function updateSegments(segments) {
+    // Update the list of segment names available
+    const shouldNotifyUpdate = storage.segments.resetSegments(segments);
+
+    // Notify update if required
+    if (storage.splits.usesSegments() && (shouldNotifyUpdate || readyOnAlreadyExistentState)) {
+      readyOnAlreadyExistentState = false;
+      segmentsEventEmitter.emit(segmentsEventEmitter.SDK_SEGMENTS_ARRIVED);
+    }
+  }
+
+  return function MySegmentsUpdater(retry = 0, segmentList) {
+    // If segmentList is provided, there is no need to fetch mySegments
+    if (segmentList) {
+      updateSegments(segmentList);
+      return;
```

Comment:
You should also return a promise on this path, otherwise you have two different types to be expected.